crispy_forms/templatetags/crispy_forms_field.py

```diff
@@ -115,13 +115,8 @@ def render(self, context):  # noqa: C901
             else:
                 css_class = class_name
 
-            if template_pack == "bootstrap4" and not is_multivalue(field):
-                if not is_checkbox(field):
-                    css_class += " form-control"
-                    if is_file(field):
-                        css_class += "-file"
-                if field.errors:
-                    css_class += " is-invalid"
+            if template_pack == "bootstrap4" and field.errors and not is_multivalue(field):
```

Comment:
We're getting there. Will be easier to split out template packs once this is gone :+1: